rasa/shared/nlu/training_data/training_data.py

```diff
@@ -739,61 +832,36 @@ def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
             )
 
             # update the data to chunk in next iteration
-            data_to_chunk = TrainingData(
-                leftover_examples,
-                responses=data_to_chunk._needed_responses_for_examples(
-                    leftover_examples
-                ),
-            )
-            all_chunks.append(
-                TrainingDataChunk(
-                    current_chunk,
-                    responses=data_to_chunk._needed_responses_for_examples(
-                        current_chunk
-                    ),
-                )
-            )
+            data_to_chunk = TrainingDataFull(leftover_examples)
+            all_chunks.append(TrainingDataChunk(current_chunk))
 
         # The last chunk is composed of whatever is left
-        all_chunks.append(
-            TrainingDataChunk(
-                data_to_chunk.training_examples, responses=data_to_chunk.responses
-            )
-        )
+        all_chunks.append(TrainingDataChunk(data_to_chunk.training_examples))
         return all_chunks
 
-    def has_e2e_examples(self):
-        """Checks if there are any training examples from e2e stories."""
-        return any(message.is_e2e_message() for message in self.training_examples)
+
+class TrainingData(TrainingDataFull):
+    """Holds loaded intent and entity training data."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes training data class."""
+        super().__init__(*args, **kwargs)
+        rasa.shared.utils.io.raise_warning(
+            f"'{self.__class__.__name__}' is deprecated and "
+            f"will be removed in the future. "
+            f"Please use the '{TrainingDataFull.__name__}' instead.",
+            category=DeprecationWarning,
+        )
 
 
-class TrainingDataChunk(TrainingData):
+class TrainingDataChunk(NLUPipelineTrainingData):
     """Holds a portion of the complete TrainingData.
 
     It can only hold training_examples and responses.
     Setting entity synonyms, regex features and lookup
     tables will result in an exception being raised.
     """
 
-    def __init__(
-        self,
-        training_examples: Optional[List[Message]] = None,
-        entity_synonyms: Optional[Dict[Text, Text]] = None,
-        regex_features: Optional[List[Dict[Text, Text]]] = None,
-        lookup_tables: Optional[List[Dict[Text, Any]]] = None,
-        responses: Optional[Dict[Text, List[Dict[Text, Any]]]] = None,
-    ) -> None:
-        """Initialize a training data chunk."""
-        if entity_synonyms or regex_features or lookup_tables:
-            raise RasaException(
-                f"{self.__class__} cannot have entity synonyms, "
-                f"regex features or lookup tables set. "
-                f"This is to reduce the memory overhead."
-            )
```

Comment:
We had discussed with Tanja that raising this exception with an informative message will ensure that this class is never instantiated with synonyms, regex, etc. and also keeps the interface the same. I do see merit in doing that. What do you think?

Reply:
but they are not in this class anymore. I thought the whole idea  of this refactor is to have different interfaces

Reply:
because chunk and full data shouldn't be allowed to use interchangeably

Reply:
Hmm, I see your point 👍